build-n-publish.sh

```diff
@@ -3,64 +3,60 @@
+# To avoid issues with MINGW y Git Bash, see:
+# https://github.com/docker/toolbox/issues/673
+# https://gist.github.com/borekb/cb1536a3685ca6fc0ad9a028e6a959e3
+export MSYS_NO_PATHCONV=1
+export MSYS2_ARG_CONV_EXCL="*"
 
-sh ./build-w-docker.sh $pkgVersion $cdnBaseUrl production $pkgVersion $pkgBuild $pkgCommitId
-docker run --rm \
-    -e AKAMAI_CDN_HOSTNAME \
-    -e AKAMAI_CDN_USERNAME \
-    -e AKAMAI_CDN_PASSWORD \
-    -e AKAMAI_CDN_CPCODE \
-    -e "PROJECT_NAME=$pkgName" \
-    -e "VERSION_NAME=$pkgVersion" \
-    -v `pwd`/build:/source \
-    dopplerrelay/doppler-relay-akamai-publish
+echo Publishing to Docker and Akamai...
+echo pkgName: $pkgName
+echo cdnBaseUrl: $cdnBaseUrl
+echo pkgVersion: $pkgVersion
+echo pkgBuild: $pkgBuild
+echo pkgCommitId: $pkgCommitId
 
-sh ./build-w-docker.sh $pkgVersion $cdnBaseUrl qa $pkgVersion $pkgBuild $pkgCommitId
-docker run --rm \
-    -e AKAMAI_CDN_HOSTNAME \
-    -e AKAMAI_CDN_USERNAME \
-    -e AKAMAI_CDN_PASSWORD \
-    -e AKAMAI_CDN_CPCODE \
-    -e "PROJECT_NAME=$pkgName" \
-    -e "VERSION_NAME=$pkgVersionQa" \
-    -v `pwd`/build:/source \
-    dopplerrelay/doppler-relay-akamai-publish
+# TODO: It could break concurrent deployments with different docker accounts
+docker login -u="$DOCKER_WEBAPP_USERNAME" -p="$DOCKER_WEBAPP_PASSWORD"
 
-sh ./build-w-docker.sh $pkgVersion $cdnBaseUrl int $pkgVersion $pkgBuild $pkgCommitId
-docker run --rm \
-    -e AKAMAI_CDN_HOSTNAME \
-    -e AKAMAI_CDN_USERNAME \
-    -e AKAMAI_CDN_PASSWORD \
-    -e AKAMAI_CDN_CPCODE \
-    -e "PROJECT_NAME=$pkgName" \
-    -e "VERSION_NAME=$pkgVersionInt" \
-    -v `pwd`/build:/source \
-    dopplerrelay/doppler-relay-akamai-publish
+# Force pull the latest image version due to the cache not always is pruned immediately after an update is uploaded to docker hub
+docker pull dopplerrelay/doppler-relay-akamai-publish
 
-sh ./build-w-docker.sh $pkgVersion $cdnBaseUrl development $pkgVersion $pkgBuild $pkgCommitId
-docker run --rm \
-    -e AKAMAI_CDN_HOSTNAME \
-    -e AKAMAI_CDN_USERNAME \
-    -e AKAMAI_CDN_PASSWORD \
-    -e AKAMAI_CDN_CPCODE \
-    -e "PROJECT_NAME=$pkgName" \
-    -e "VERSION_NAME=$pkgVersionDevelopment" \
-    -v `pwd`/build:/source \
-    dopplerrelay/doppler-relay-akamai-publish
+for environment in ${environments}; do
+    echo Publishing ${environment}...
+
+    if test "$environment" = "production";
+    then  
+        env_version=$pkgVersion;
+    else
+        env_version=$environment-$pkgVersion;
+    fi
+
+    echo deploying into $env_version folder ...
+
+    docker build --pull \
+        -t darosw/doppler-webapp:$environment \
+        --build-arg environment=$environment \
+        --build-arg env_version=$env_version \
+        --build-arg cdn_hostname=$AKAMAI_CDN_HOSTNAME \
+        --build-arg cdn_username=$AKAMAI_CDN_USERNAME \
+        --build-arg cdn_password=$AKAMAI_CDN_PASSWORD \
+        --build-arg cdn_cpcode=$AKAMAI_CDN_CPCODE \
+        -f Dockerfile.BUILDS \
```

Comment:
What do you think about: 

```suggestion
        -f Dockerfile.BUILDS_AND_CDN \
```

Or



```suggestion
        -f Dockerfile.PULLREQUESTS \
```

Reply:
Sounds nice.